Mathlib/Computability/ContextFreeGrammar.lean

```diff
@@ -194,6 +206,28 @@ lemma Derives.append_right {v w : List (Symbol T g.NT)}
   | refl => rfl
   | tail _ last ih => exact ih.trans_produces <| last.append_right p
 
+lemma produces_nonterminal (g : ContextFreeGrammar T)
```

Comment:
I'd rather see `g` implicit (and perhaps `u` and `v` implicit as well) and `h` renamed to `hg` or `hguv` here.